### DISCUSSION TITLE
1.1.2b4 - Sidechaining routing question

### DISCUSSION THREAD

**Discussion by addsubtractstudios** [apohl79/audiogridder#666] [Ideas]
Hiya, firstly, amazing app. Running some tests in Audio Hijack in a local setup, before I attempt to integrate it into the workflow and add an old yet powerful pc as a Server on the network. Very impressed so far!

I've been using UAD Luna for the past year (long time Logic user prior), and currently, it doesn't support Sidechaining/Key Input.
As a note for clarification, UAD Luna supports 3rd party AU plugs, some of which support Sidechaining, however, Luna doesn't offer a sidechain "bus" so the function isn't used within these plugins.

My question; 
   Is AG capable of allowing or could it offer this functionality? I know that a plugin within a DAW basically looks for an SC/Key 'bus' to then pass that audio to the plug to utilize, but is there any way to Hijack a signal from somewhere to feed the SC input to the server? I understand AG is pulling audio this buss from the Daw but is there any way to potentially feed the active inputs (of the AG server Sidechain Channels) with a feed of my choice? I thought came close with BCA's Patchworks. As in AG, I can see the SC in/outs as channels but I've found no method to patch audio into them except via a Daw.

Needless to say, I've tested the AG's SC in Logic (with FF-MB & ProQ3) and it works a treat as expected.

If it's a workable angle then I know a very large group of UA users that would be on it in a heartbeat!! A number of users need SC for frequency-based processing, a little more complex at times than Kick and Bass Sidechaining, which can be easily worked around in Luna with midi tools such as Shaperbox, or LFO tools.

Luna, like other Daws I've noted in the threads here, also struggles with CPU/core allocation. The local functionality of AG is brilliant btw. Looking forward to pushing the envelope of my unused cores with some more testing tonight!

Thanks,
Adam.

**Comment by apohl79** [comment]
Hey Adam,

interesting. Never thought about it, but to support, what you are asking, AG would need some sort of SC receive plugin, that you can insert into any channel and which just passes audio through. This plugin could than tap the audio of that channel and could send it to an AG FX plugin as separate input. 

This is not supported today, as AG can only receive and process the DAW channel audio provided to the plugin. If the DAW supports sidechains, it just provides another input bus to the plugin.

That said, it is doable, but not a small change. I will create a feature request ticket for this.

Cheers,
Andreas

**Reply by addsubtractstudios** [reply]
Thanks for the reply Andres, and for consideration as a feature - Would be very cool to utilize SC within any AG instances.

I did some more digging around with a better understanding of the background routings of sidechain buses, and I actually found a product that does the job (almost) perfectly. 'Metaplugin' by DDMF - It does it in exactly the way you mentioned, with a send/receive plugin that can port audio from a channel of your choice into a receive instance of the same plugin. I'm amazed I never came across it in my hunting and years of mixing as it's actually a damn useful tool!! It's very much like 'Element' (almost the identical Juce interface!) - but a little more versatile and it has the 'Sendit' plugin as a part of the package making the routing possibilities much greater. The only negative is that the 'Sendit' plugin doesn't seem to offer PDC, throwing 512 samples of latency into the SC path (luna's fixed buffer). Not a major and an easy workaround so I've got something to work for now :-)

I would love to see a similar 'TX/RX' plugin in AG still, having the option to shift things over to Server, including SC support is something I'm finding more and more invaluable as the workload increases on it!

Thanks again,

Adam.
